matrix_client/api.py

```diff
@@ -107,34 +107,21 @@ def sync(self, since=None, timeout_ms=30000, filter=None,
     def validate_certificate(self, valid):
         self.validate_cert = valid
 
-    def register(self, content=None, kind='user'):
+    def register(self, auth_body, kind, **kwargs):
```

Comment:
Let's explicitly list all the possible kwargs here. `bind_email`, `username`, `password` `device_id`, and `initial_device_display_name`.

Reply:
Not sure if it was clear before. I want the following listed in the function signature; not just the docstring. :)

Reply:
Wouldn't that defeat the point of using kwargs ?
I think you mean a function signature like `register(self, auth_body, bind_email=None, ...)`

Reply:
Yep. I think this method would be better without kwargs with each possible top-level key listed explicitly. Just like you showed, yes.